src/IRVisualization.h

```diff
@@ -0,0 +1,189 @@
+#ifndef IRVISUALIZATION_H
+#define IRVISUALIZATION_H
+
+#include "FindStmtCost.h"
+#include "IRVisitor.h"
+
+#include <set>
+#include <unordered_map>
+
+using namespace std;
+using namespace Halide;
+using namespace Internal;
+
+#define MAX_CONDITION_LENGTH 35
+#define MAX_NUMBER_OF_NODES 15
+
+#define NUMBER_COST_COLORS 20
+
+struct StmtSize {
+    map<string, string> writes;
+    map<string, string> reads;
+
+    bool empty() const {
+        return writes.size() == 0 && reads.size() == 0;
+    }
+};
+
+/*
```

Comment:
It would be great to do the doxygen style comments so the classes show up in the documentation.